File: payola/web/public/javascripts/cz/payola/web/client/views/graph/VertexView.js
goog.provide('cz.payola.web.client.views.graph.VertexView');
goog.require('cz.payola.web.client.views.Constants');
goog.require('cz.payola.web.client.views.Point');
goog.require('cz.payola.web.client.views.graph.InformationView');
goog.require('cz.payola.web.client.views.graph.View');
cz.payola.web.client.views.graph.VertexView = function(vertexModel, position) {
var self = this;
self.vertexModel = vertexModel;
self.position = position;
self.selected = false;
self.information = new cz.payola.web.client.views.graph.InformationView(self.vertexModel);
goog.base(self);};
goog.inherits(cz.payola.web.client.views.graph.VertexView, cz.payola.web.client.views.graph.View);
cz.payola.web.client.views.graph.VertexView.prototype.draw = function(context, color, positionCorrection) {
var self = this;
var correction = (function() {
if ((positionCorrection != null)) {
return positionCorrection.toVector();
} else {
return cz.payola.web.client.views.Point.Zero.toVector();
}})();
self.drawRoundedRectangle(context, self.position.$plus(cz.payola.web.client.views.Constants.VertexSize.$div(-2.0)).$plus(correction), cz.payola.web.client.views.Constants.VertexSize, cz.payola.web.client.views.Constants.VertexCornerRadius);
context.fillStyle = (function() {
if ((color != null)) {
return color.toString();
} else {
return cz.payola.web.client.views.Constants.ColorVertexDefault.toString();
}})();
context.fill();
};
cz.payola.web.client.views.graph.VertexView.prototype.metaClass_ = new s2js.MetaClass('cz.payola.web.client.views.graph.VertexView', [cz.payola.web.client.views.graph.View]);

File: payola/web/client/src/main/scala/cz/payola/web/client/views/graph/VertexView.scala
package cz.payola.web.client.views.graph

import s2js.adapters.js.dom.CanvasRenderingContext2D
import cz.payola.web.client.views.Constants._
import cz.payola.web.client.views.{Color, Point}
import collection.mutable.ListBuffer
import cz.payola.common.rdf.{IdentifiedVertex, Vertex}

class VertexView(val vertexModel: Vertex, var position: Point) extends View {
    var selected = false

    var edges = ListBuffer[EdgeView]()

    val information: Option[InformationView] = vertexModel match {
        case i: IdentifiedVertex => Some(new InformationView(i))
        case _ => None
    }

    def draw(context: CanvasRenderingContext2D, color: Option[Color], positionCorrection: Option[Point]) {
        
        val correction = positionCorrection match {
            case None => Point.Zero.toVector
            case _ => positionCorrection.get.toVector
        }

        drawRoundedRectangle(context, this.position + (VertexSize / -2) + correction, VertexSize, VertexCornerRadius)

        val colorToUse = color match {
            case None => ColorVertexDefault
            case _ => color.get
        }

        fillCurrentSpace(context, colorToUse)
    }
}